File: 2020/day02.rb
data = File.read(File.basename(__FILE__, ".rb") + ".input")

# data = <<~DATA
# 1-3 a: abcde
# 1-3 b: cdefg
# 2-9 c: ccccccccc
# DATA

puts data.split("\n").count { |l|
  min, max, l, pw = l.chomp.split(/[\- :]+/)
  (min.to_i..max.to_i).include?(pw.count(l))
}

puts data.split("\n").count { |l|
  min, max, l, pw = l.chomp.split(/[\- :]+/)
  (pw[min.to_i-1] == l) ^ (pw[max.to_i-1] == l)
}

File: 2017/day02.py
from inspect import cleandoc
from itertools import combinations

import os
import math

test_data = cleandoc("""
5 1 9 5
7 5 3
2 4 6 8
""")

test_data = cleandoc("""
5 9 2 8
9 4 7 3
3 8 6 5
""")

base = os.path.splitext(os.path.basename(__file__))[0]
test_data = open(base + ".input").read()

total = 0
total_div = 0
for row in test_data.splitlines():
    ints = list(map(int, row.split()))
    total += (max(ints) - min(ints))
    total_div += sum((y // x) for x, y in list(combinations(sorted(ints), 2)) if y % x == 0)

print(total)
print(total_div)
